vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java

```diff
@@ -120,8 +120,8 @@ public void setTarget(Component target) {
         if (getTarget() != null) {
             targetBeforeOpenRegistration.remove();
             targetAttachRegistration.remove();
-            getTarget().getElement().callJsFunction(
-                    "$contextMenuTargetConnector.removeConnector");
+            getTarget().getElement().executeJs(
+                    "if (this.$contextMenuTargetConnector) { this.$contextMenuTargetConnector.removeConnector() }");
```

Comment:
It seems the problem is that the JS call is only actually run after a label has been added to the page again, at which point the connector instance doesn't exist anymore because the DOM element is a new instance.

For example:
- Change target to label 2
- Schedules JS to remove connector for label 1
- Label 1 is removed
- JS doesn't run, because element isn't attached, but JS call also isn't discarded
- ...
- Add label 1 again
- JS call runs

While the fix here would be OK, I wonder if there isn't a better way to handle this? Should Flow discard JS calls when an element is removed, since those are not part of the element's state? Is there already a way to do so without making framework users write complex code with checking for detach etc.?

Reply:
Let's go ahead with the proposed fix for now.